src/Autofac/Core/Resolving/CircularDependencyDetector.cs

```diff
@@ -61,8 +61,11 @@ public static void CheckForCircularDependency(IComponentRegistration registratio
                 throw new DependencyResolutionException(string.Format(CultureInfo.CurrentCulture, CircularDependencyDetectorResources.MaxDepthExceeded, registration));
 
             // Checks for circular dependency
-            if (activationStack.Any(a => a.ComponentRegistration == registration))
-                throw new DependencyResolutionException(string.Format(CultureInfo.CurrentCulture, CircularDependencyDetectorResources.CircularDependency, CreateDependencyGraphTo(registration, activationStack)));
+            foreach (var a in activationStack)
```

Comment:
Nice

Reply:
I brought all the changes from this file into the `develop` branch.